### PR TITLE
feat: introduced time component

### LINKED ISSUE #5481: Create <Time/> component
Create a generic `Time` cimponent that displays date info in the curent locale.

The component should have two props: `date`, which should be a string or `Date` object and the second one. corresponding to the shape of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#options

The component should return a `time` element, with the DateTime formatted on the currently applied locale (so the component must use `useLocale` hook).

See the repeated usage of a workaround from https://github.com/nodejs/nodejs.org/pull/5479. The new component would replace these.

```jsx
<time dateTime={post.date}>
  {new Date(post.date).toLocaleString('en-GB', {
    timeZone: 'UTC',
    month: 'short',
    day: '2-digit',
  })}
</time>
```

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR introduces a simple `Time` component that fulfils the specification presented on the issue.

It also updates the `en` locale to use `en-GB`

## Related Issues

Fixes #5481

### CODE REVIEW

**Review by AugustinMauroy — CHANGES_REQUESTED** [review]
You are recoding an fonctionnality of react-intl (formattedDate)

**Comment by ovflowd** [comment]
> You are recoding an fonctionnality of react-intl (formattedDate)

I'm not. Why do you think this component is reimplementing existing functionality from `react-intl`?

Edit: I see your point; refer to my comment on https://github.com/nodejs/nodejs.org/issues/5481#issuecomment-1627720784

**Comment by ovflowd** [comment]
I'm updating this PR to use `FormattedDate`, and I want to clarify that we're not "redoing/reimplementing" existing functionality. It's often more beneficial to utilize Native JavaScript APIs instead of relying on sugarcoated components.

Additionally, it's important to note that the `Time` component acts as a "wrapper" for HTML's `time` element. It doesn't replace `react-intl`'s Date & Time Components, which return a plain JSX-string.

I'm providing these explicit details to ensure you understand why your assessment may not be accurate, even though using `FormattedDate` makes sense in this case.

File: components/Common/Time/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { useMemo } from 'react';
+import { useLocale } from '@/hooks/useLocale';
+import type { FC } from 'react';
+
+type TimeProps = { date: string | Date; format: Intl.DateTimeFormatOptions };
+
+export const Time: FC<TimeProps> = ({ date, format }) => {
+  const { currentLocale } = useLocale();
+
+  const { isoDate, formatedDate } = useMemo(() => {
```
Comment (mikeesto) [inline components/Common/Time/index.tsx]:
```suggestion
  const { isoDate, formattedDate } = useMemo(() => {
```

File: components/Common/Time/index.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    const dateObject = new Date(date);
+
+    return {
+      isoDate: dateObject.toISOString(),
+      formatedDate: dateObject.toLocaleString(currentLocale.hrefLang, {
```
Comment (mikeesto) [inline components/Common/Time/index.tsx]:
```suggestion
      formattedDate: dateObject.toLocaleString(currentLocale.hrefLang, {
```

File: components/Common/Time/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        timeZone: 'UTC',
+      }),
+    };
+    // We assume that the `format` options will not change and be static
+    // and we calculate the date difference by comparing its string value
+    // since its ISO format is what matters for us the most.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [date.toString(), currentLocale.hrefLang]);
+
+  return <time dateTime={isoDate}>{formatedDate}</time>;
```
Comment (mikeesto) [inline components/Common/Time/index.tsx]:
```suggestion
  return <time dateTime={isoDate}>{formattedDate}</time>;
```

**Review by mikeesto — APPROVED** [review]

**Pull request closed without merging**